src/components/Tabs/Tabs.vue

```diff
@@ -0,0 +1,562 @@
+    containerWidth,
+    disclosureWidth,
+  } = measurements;
+
+  const { visibleTabs, hiddenTabs } = getVisibleAndHiddenTabIndices(
+    props.tabs,
+    props.selected,
+    disclosureWidth,
+    tabWidths,
+    containerWidth,
+  );
+
+  state.visibleTabs = visibleTabs;
+  state.hiddenTabs = hiddenTabs;
+  state.disclosureWidth = disclosureWidth;
+  state.containerWidth = containerWidth;
+  state.tabWidths = tabWidths;
+};
+
+const handleListTabClick = (id: string) => {
+  handleTabClick(id);
+  handleClose();
+  state.isTabsFocused = true;
+};
+
+const moveToActiveTab = (offsetLeft: number) => {
+  setTimeout(() => {
+    if (scrollRef.value && typeof scrollRef.value.scroll === 'function') {
+      const scrollRefOffset = wrapRef?.value?.offsetLeft || 0;
+      scrollRef?.value?.scroll({
+        left: offsetLeft - scrollRefOffset,
+      });
+    }
+  }, 0);
+};
+
+const handleTabFocus = () => {
+  if (state.modalSubmitted) {
+    state.tabToFocus = props.selected;
+    state.modalSubmitted = false;
+  }
+};
+
+onMounted(() => {
+  prevModalOpen.value = state.isTabModalOpen;
+  prevPopoverOpen.value = state.isTabPopoverOpen;
+});
+
+watch(
+  () => [
+    prevPopoverOpen.value,
+    state.isTabPopoverOpen,
+    prevModalOpen.value,
+    state.isTabModalOpen,
+    props.selected,
+    state.tabToFocus,
+  ],
+  () => {
+    const hasModalClosed = prevModalOpen.value && !state.isTabModalOpen;
+    const hasPopoverClosed = prevPopoverOpen.value && !state.isTabPopoverOpen;
+    if (hasModalClosed) {
+      state.isTabsFocused = true;
+      state.tabToFocus = props.selected;
+    } else if (hasPopoverClosed && !state.isTabModalOpen) {
+      state.isTabsFocused = true;
+      state.tabToFocus = props.selected;
+    }
+  },
+);
+
+watch(
+  () => [
+    state.containerWidth,
+    state.disclosureWidth,
+    props.tabs,
+    props.selected,
+    state.tabWidths,
+  ],
+  () => {
+    const { visibleTabs, hiddenTabs } = getVisibleAndHiddenTabIndices(
+      props.tabs,
+      props.selected,
+      state.disclosureWidth,
+      state.tabWidths,
+      state.containerWidth,
+    );
+
+    state.visibleTabs = visibleTabs;
+    state.hiddenTabs = hiddenTabs;
+  },
+  { immediate: true }
+);
+
+watch(
+  () => [
+    props.selected,
+    breakpoints.value.mdDown,
+  ],
+  () => {
+    if (breakpoints.value.mdDown) {
+      moveToSelectedTab();
+    }
+  },
+);
+
+watch(
+  () => [
+    state.isTabsFocused,
+    props.selected,
+    state.showDisclosure,
+  ],
+  () => {
+    if (state.isTabsFocused && !state.showDisclosure) {
+      state.tabToFocus = props.selected;
+    }
+  },
+);
+
+const activator = () => h(
+  resolveComponent('UnstyledButton'),
+  {
+    type: 'button',
+    className: classNames(styles.DisclosureActivator),
+    disabled: props.disabled,
+    'aria-label': props.disclosureText ?? i18n.translate('Polaris.Tabs.toggleTabsLabel'),
+    onClick: handleDisclosureActivatorClick,
+  },
+  () => [
+    h(
+      'template',
```

Comment:
?

Reply:
has warning `Non-function value encountered for default slot`